packages/CopyInput/src/CopyInput.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Input from "@paprika/input";
+import Popover from "@paprika/popover";
+import Button from "@paprika/button";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import extractChildrenProps from "@paprika/helpers/lib/extractChildrenProps";
+import CopyIcon from "@paprika/icon/lib/ArrowDown"; // TODO: Replace with actual icon when ready
+import CopyInputInputPropsCollector from "./CopyInputInputPropsCollector";
+import * as sc from "./CopyInput.styles";
+
+function CopyInput(props) {
+  const { defaultValue, children, ...moreProps } = props;
+  const I18n = useI18n();
+  const extendedInputProps = extractChildrenProps(children, CopyInputInputPropsCollector);
+
+  const [value, setValue] = React.useState(defaultValue);
+  const [isClickedTooltipOpen, setIsClickedTooltipOpen] = React.useState(false);
+
+  function handleButtonClick() {
+    navigator.clipboard.writeText(value);
+    setIsClickedTooltipOpen(true);
+    setTimeout(() => setIsClickedTooltipOpen(false), 5000);
+  }
+
+  return (
+    <sc.CopyInput data-pka-anchor="copy-input" {...moreProps}>
+      <Input defaultValue={defaultValue} onChange={e => setValue(e.target.value)} {...extendedInputProps} />
+      <Popover
+        align="bottom"
+        isDark={!isClickedTooltipOpen}
+        isEager={!isClickedTooltipOpen}
+        isOpen={isClickedTooltipOpen || null}
```

Comment:
why `|| null` instead of `|| false`? Shouldn't `isOpen` be a boolean?

Reply:
It needs to be `null` or else the popover won't appear on hover (from `isEager`). This is a bit of a weird popover because it needs to render one way on hover, and another way on click.

Reply:
It _is_ a tricky one! Currently the `isEager` version of the tooltip is wrapping the `<Button>` below with a `<RawButton>` (which isn't ideal), because the `<Popover.Trigger>` doesn't have a function for a children.  Probably you'll want to use that render prop pattern.  

I recently revamped the Popover Storybook stories – I hope they now do a better job of explaining how the compoent works, and with clearer code examples. 

https://paprika.highbond.com/?path=/story/messaging-popover-docs--page